homeassistant/components/egauge/sensor.py

```diff
@@ -0,0 +1,110 @@
+"""Sensor platform for eGauge energy monitors."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from egauge_async.json.models import RegisterType
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfEnergy, UnitOfPower
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .coordinator import EgaugeConfigEntry, EgaugeDataCoordinator
+from .entity import EgaugeEntity
+from .models import EgaugeData
+
+
+@dataclass(frozen=True, kw_only=True)
+class EgaugeSensorEntityDescription(SensorEntityDescription):
+    """Extended sensor description for eGauge sensors."""
+
+    native_value_fn: Callable[[EgaugeData, str], float | None]
+
+
+POWER_SENSOR = EgaugeSensorEntityDescription(
+    key="power",
+    translation_key="power",
+    device_class=SensorDeviceClass.POWER,
+    state_class=SensorStateClass.MEASUREMENT,
+    native_unit_of_measurement=UnitOfPower.WATT,
+    native_value_fn=lambda data, register: data.measurements.get(register),
+)
+
+ENERGY_SENSOR = EgaugeSensorEntityDescription(
+    key="energy",
+    translation_key="energy",
+    device_class=SensorDeviceClass.ENERGY,
+    state_class=SensorStateClass.TOTAL_INCREASING,
+    native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+    native_value_fn=lambda data, register: (
+        data.counters[register] / 3_600_000
```

Comment:
Ideally we set the native UoM to the native unit, as in, let's try to avoid to divide by 3600000 ourselves and just get to the closest unit for UnitOfEnergy, and use suggested_unit_of_measurement to set it to kWh